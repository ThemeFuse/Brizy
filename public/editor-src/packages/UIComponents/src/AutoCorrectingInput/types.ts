import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
