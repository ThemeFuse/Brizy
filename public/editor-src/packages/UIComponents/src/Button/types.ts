import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
