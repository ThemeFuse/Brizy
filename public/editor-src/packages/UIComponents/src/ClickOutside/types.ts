import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
