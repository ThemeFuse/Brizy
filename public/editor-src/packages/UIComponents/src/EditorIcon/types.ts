export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
