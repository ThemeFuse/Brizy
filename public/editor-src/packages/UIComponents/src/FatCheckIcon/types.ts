import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
