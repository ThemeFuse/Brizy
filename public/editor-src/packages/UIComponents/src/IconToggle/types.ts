import { ReactElement } from "react";
import { NonEmptyArray } from "../types/array";
import { WithClassName, WithOnChange, WithValue } from "../types/attributes";
import { Props as ItemProps } from "./IconToggleItem";

export type Props<T> = WithClassName &
  WithValue<T> &
  WithOnChange<T> & {
    children: NonEmptyArray<ReactElement<ItemProps<T>>>;
  };
