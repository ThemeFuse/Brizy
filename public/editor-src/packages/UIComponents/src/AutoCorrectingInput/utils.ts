import { clamp } from "../utils/math";

export function formatInputValue(value: number, step: number): string | number {
  const stepDecimalLength = decimalLength(step);

  return stepDecimalLength > 0
    ? value.toFixed(stepDecimalLength)
    : String(value);
}

export function decimalLength(number: number): number {
  const [, decimal] = String(number).split(".");
  return decimal ? decimal.length : 0;
}

export function correctNumber(
  number: number,
  min: number,
  max: number,
  step: number
): number {
  return roundByStep(clamp(number, min, max), step, min);
}

export const preciseAdd = preciseOperation.bind(null, "add");

export const preciseSub = preciseOperation.bind(null, "sub");

function roundByStep(number: number, step: number, offset = 0) {
  // this is done to avoid quirks with floating point operations
  // like 1.1 - 1 = 0.10000000000000009
  const mul = 10 ** decimalLength(step);

  number = number * mul;
  step = step * mul;
  offset = offset * mul;

  const tmp = Math.ceil((number - offset) / step) * step + offset;

  return tmp / mul;
}

function preciseOperation(type: "add" | "sub", a: number, b: number) {
  const dla = decimalLength(a);
  const dlb = decimalLength(b);
  const dlMax = Math.max(dla, dlb);
  const mul = 10 ** dlMax;

  const a_ = a * mul;
  const b_ = b * mul;
  let res;

  switch (type) {
    case "add":
      res = a_ + b_;
      break;
    case "sub":
      res = a_ - b_;
      break;
    default:
      throw new Error(`operation type unknown ${type}`);
  }

  return res / mul;
}
