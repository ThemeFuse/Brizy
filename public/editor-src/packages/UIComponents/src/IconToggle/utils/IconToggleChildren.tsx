import React, { ReactElement } from "react";
import { IconsName } from "../../EditorIcon/types";
import { NonEmptyArray } from "../../types/array";
import { Literal } from "../../types/literal";
import { IconToggleItem, Props as ItemProps } from "../IconToggleItem";

const choices = [
  { icon: IconsName.TextAlignLeft, title: "Align left", value: "left" },
  { icon: IconsName.TextAlignCenter, title: "Align center", value: "center" },
  { icon: IconsName.TextAlignRight, title: "Align right", value: "right" }
];

export const children = choices.map(({ icon, value, title }, i) => (
  <IconToggleItem<Literal> key={i} value={value} icon={icon} title={title} />
)) as NonEmptyArray<ReactElement<ItemProps<Literal>>>;
