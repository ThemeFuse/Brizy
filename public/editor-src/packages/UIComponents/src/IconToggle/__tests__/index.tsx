import { render, waitFor } from "@testing-library/react";
import React from "react";
import { Literal } from "../../types/literal";
import { IconToggle } from "../index";
import { children } from "../utils/IconToggleChildren";

describe("Test IconToggle", () => {
  const onChange = jest.fn();

  describe("Snapshots", () => {
    test("Render left Icon", async () => {
      const iconToggle = render(
        <IconToggle<Literal> onChange={onChange} value="left">
          {children}
        </IconToggle>
      );

      await waitFor(() => {
        expect(iconToggle.container.firstChild).toBeInTheDocument();
      });
      expect(iconToggle.container.firstChild).toMatchSnapshot();
    });
    test("Render Center Icon", async () => {
      const iconToggle = render(
        <IconToggle<Literal> onChange={onChange} value="center">
          {children}
        </IconToggle>
      );

      await waitFor(() => {
        expect(iconToggle.container.firstChild).toBeInTheDocument();
      });
      expect(iconToggle.container.firstChild).toMatchSnapshot();
    });
    test("Render Right Icon", async () => {
      const iconToggle = render(
        <IconToggle<Literal> onChange={onChange} value="right">
          {children}
        </IconToggle>
      );
      await waitFor(() => {
        expect(iconToggle.container.firstChild).toBeInTheDocument();
      });
      expect(iconToggle.container.firstChild).toMatchSnapshot();
    });
  });
});
