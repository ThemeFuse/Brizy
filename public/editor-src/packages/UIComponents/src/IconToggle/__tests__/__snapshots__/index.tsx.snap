// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test IconToggle Snapshots Render Center Icon 1`] = `
<div
  class="brz-ed-control__icon-carousel"
  title="Align center"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial]"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
      >
        <path
          d="M0 1h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M3 5h10v2H3z"
          data-color="color-2"
        />
        <path
          d="M0 9h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M3 13h10v2H3z"
          data-color="color-2"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Test IconToggle Snapshots Render Right Icon 1`] = `
<div
  class="brz-ed-control__icon-carousel"
  title="Align right"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial]"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
      >
        <path
          d="M0 1h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M6 5h10v2H6z"
          data-color="color-2"
        />
        <path
          d="M0 9h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M6 13h10v2H6z"
          data-color="color-2"
        />
      </g>
    </g>
  </svg>
</div>
`;

exports[`Test IconToggle Snapshots Render left Icon 1`] = `
<div
  class="brz-ed-control__icon-carousel"
  title="Align left"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial]"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g
        fill="currentColor"
      >
        <path
          d="M0 1h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M0 5h10v2H0z"
          data-color="color-2"
        />
        <path
          d="M0 9h16v2H0z"
          fill="currentColor"
        />
        <path
          d="M0 13h10v2H0z"
          data-color="color-2"
        />
      </g>
    </g>
  </svg>
</div>
`;
